crates/ironrdp-web/src/session.rs

```diff
@@ -216,13 +205,28 @@ impl SessionBuilder {
         self.clone()
     }
 
-    /// Optional
-    pub fn use_display_control(&self) -> SessionBuilder {
-        self.0.borrow_mut().use_display_control = true;
+    pub fn extension(&self, value: JsValue) -> SessionBuilder {
+        match serde_wasm_bindgen::from_value::<Extension>(value) {
+            Ok(value) => match value {
+                Extension::KdcProxyUrl(kdc_proxy_url) => self.0.borrow_mut().kdc_proxy_url = Some(kdc_proxy_url),
+                Extension::Pcb(pcb) => self.0.borrow_mut().pcb = Some(pcb),
+                Extension::DisplayControl(use_display_control) => {
+                    self.0.borrow_mut().use_display_control = use_display_control
+                }
+            },
+            Err(error) => error!(%error, "Provided JsValue is not a valid extension value"),
+        }
+
+        self.clone()
+    }
+
+    pub fn extension_call(&self, _ident: String, _call: js_sys::Function) -> SessionBuilder {
+        warn!("Extension calls are not supported yet");
+
         self.clone()
     }
```

Comment:
question: In the design I had, `extension_call` was only part of `Session`, not `SessionBuilder`. Do you have a use case for `extension_call` on `SessionBuilder`?